test/CMakeLists.txt

```diff
@@ -40,11 +40,13 @@ function(tbb_add_test)
         ${TBB_COMMON_COMPILE_FLAGS}
     )
 
+    if (UNIX)
+        target_link_libraries(${_tbb_test_TARGET_NAME} PRIVATE -rdynamic) # for the test_dynamic_link
```

Comment:
According to the comment `-rdynamic` is needed only for `test_dynamic_link`. Why we add it for all the tests? @alexey-katranov, @pavelkumbrasev any ideas?
Probably it was just copy-pasting. If no reason to add this flag for all the tests I'd propose to use more precise approach with adding this flag just for `test_dynamic_link`.

BTW, condition `if (UNIX)` will lead to duplication of this option on Linux, because it is passed by default there:
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Linux-GNU.cmake#L14
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Linux-Intel.cmake#L37

Not a big deal really, but maybe use condition like `if (APPLE OR ANDROID_PLATFORM)`.

Having `-rdynamic` default on Linux makes my comments above just style-related.